src/tests/simulation/src/cluster.rs

```diff
@@ -891,27 +891,15 @@ impl Cluster {
 #[cfg_or_panic(madsim)]
 impl Drop for Cluster {
     fn drop(&mut self) {
-        // FIXME: remove it when deprecate the on-disk version.
-        let default_path = PathBuf::from(".");
-        let sqlite_data_dir = self
-            .config
-            .sqlite_data_dir
-            .as_ref()
-            .unwrap_or_else(|| &default_path);
-        for entry in std::fs::read_dir(sqlite_data_dir).unwrap() {
-            let entry = entry.unwrap();
-            let path = entry.path();
-            if path
-                .file_name()
-                .unwrap()
-                .to_str()
-                .unwrap()
-                .starts_with(&format!("stest-{}-", self.handle.seed()))
-            {
-                std::fs::remove_file(path).unwrap();
-                break;
-            }
-        }
+        let file_path = {
+            let mut file_path = Default::default();
+            SQLITE_PATH.with(|p| {
+                file_path = p.clone();
+            });
+            file_path
+        };
```

Comment:
Why not use `tempfile::tempdir()`? 🤔

Reply:
I actually wanted to use that. See: https://github.com/risingwavelabs/risingwave/pull/18970.
I was gonna abandon it because it instantiated NamedTempFile statically, and the destructor wouldn't be called.

But I just pushed a commit to move it inside the `Cluster's` fields instead, so it lives as long as the cluster, and will only be dropped when cluster is dropped.

Will see if that works. This PR is just a quick fix. We can use the other PR if it passes CI.